tests/apollo/test_skvbc_byzantine_primary_preexecution.py

```diff
@@ -0,0 +1,396 @@
+# Concord
+#
+# Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+#
+# This product is licensed to you under the Apache 2.0 license (the "License").
+# You may not use this product except in compliance with the Apache 2.0 License.
+#
+# This product may include a number of subcomponents with separate copyright
+# notices and license terms. Your use of these subcomponents is subject to the
+# terms and conditions of the subcomponent's license, as noted in the LICENSE
+# file.
+import os.path
+import random
+import unittest
+from os import environ
+
+import trio
+
+from util import skvbc as kvbc
+from util.bft import with_trio, with_bft_network, KEY_FILE_PREFIX, with_constant_load
+from util.skvbc_history_tracker import verify_linearizability
+import util.eliot_logging as log
+
+SKVBC_INIT_GRACE_TIME = 2
+BATCH_SIZE = 4
+NUM_OF_SEQ_WRITES = 25
+NUM_OF_PARALLEL_WRITES = 100
+MAX_CONCURRENCY = 10
+LONG_REQ_TIMEOUT_MILLI = 15000
+ASSERT_TIMEOUT_SEC = 90
+
+def start_replica_cmd(builddir, replica_id, view_change_timeout_milli="10000"):
+    """
+    Return a command that starts an skvbc replica when passed to
+    subprocess.Popen.
+    The replica is started with a short view change timeout.
+    Note each arguments is an element in a list.
+    """
```

Comment:
can you please document what is the byzantine behavior if this function is used for starting replicas?

Reply:
I will document the Byzantine behaviour for all test cases. After you pointed out, I found its missing for all the functions. Thanks @teoparvanov